scipy/datasets/tests/test_data.py

```diff
@@ -19,37 +22,52 @@ def _has_hash(path, expected_hash):
     return pooch.file_hash(path) == expected_hash
 
 
-def test_download_all():
-    # This test requires INTERNET CONNECTION
-    data_dir = data.path
-    for dataset in registry:
-        data.fetch(dataset)
+class TestDatasets:
+
+    @pytest.fixture(scope='module', autouse=True)
+    def test_download_all(self):
+        # This fixture requires INTERNET CONNECTION
+
+        # test_setup phase
+        for dataset in registry:
+            data.fetch(dataset)
+
+        yield
 
-    assert len(os.listdir(data_dir)) >= len(registry)
```

Comment:
@rgommers As this was converted to a fixture, I had to move the assert out into a new `test_existence_all` because it is not a good practice to add asserts during test_setup or test_teardown phase inside a fixture.